include/bl_storage.h

```diff
@@ -132,39 +131,13 @@ int public_key_data_read(uint32_t key_idx, uint8_t *p_buf);
  */
 void invalidate_public_key(uint32_t key_idx);
 
-/**
- * @brief Get the number of monotonic counter slots.
- *
- * @return The number of slots. If the provision page does not contain the
- *         information, 0 is returned.
- */
-uint16_t num_monotonic_counter_slots(void);
-
-/**
- * @brief Get the current HW monotonic counter.
- *
- * @return The current value of the counter.
- */
-uint16_t get_monotonic_counter(void);
-
-/**
- * @brief Set the current HW monotonic counter.
- *
- * @note FYI for users looking at the values directly in flash:
- *       Values are stored with their bits flipped. This is to squeeze one more
- *       value out of the counter.
- *
- * @param[in]  new_counter  The new counter value. Must be larger than the
- *                          current value.
- *
- * @retval 0        The counter was updated successfully.
- * @retval -EINVAL  @p new_counter is invalid (must be larger than current
- *                  counter, and cannot be 0xFFFF).
- * @retval -ENOMEM  There are no more free counter slots (see
- *                  @kconfig{CONFIG_SB_NUM_VER_COUNTER_SLOTS}).
+/* In case this gets included in an image that is build without a bootloader
+ * the provision area is not defined, so we have to point to the OTP manually
+ * NCSDK-18032 is to prevent this workaround in the future
  */
-int set_monotonic_counter(uint16_t new_counter);
-
+#ifndef PM_PROVISION_ADDRESS
+#define PM_PROVISION_ADDRESS &(NRF_UICR->OTP)
+#endif
```

Comment:
Let's move this closer to the BL_STORAGE define above.

Reply:
I see this is fixed in a later commit.

It's unfortunate that this patch is in this commit :(